src/differential_arithmetic.jl

```diff
@@ -38,14 +38,10 @@ for T in (:One, :AbstractThunk, :Any)
     @eval Base.:+(::Zero, b::$T) = b
     @eval Base.:+(a::$T, ::Zero) = a
 
-    if T !== :Any
-        @eval Base.:*(::Zero, ::$T) = Zero()
-        @eval Base.:*(::$T, ::Zero) = Zero()
-    end
+    @eval Base.:*(::Zero, x::$T) = zero(x)
+    @eval Base.:*(x::$T, ::Zero) = zero(x)
 end
-Base.:*(::Zero, x) = zero(x)
-Base.:*(x, ::Zero) = zero(x)
-
+Base.zero(::AbstractDifferential) = Zero()
```

Comment:
This should be defined in https://github.com/JuliaDiff/ChainRulesCore.jl/blob/master/src/differentials/zero.jl

and should probably have tests.

Reply:
That would make later revise harder though. Maybe just use a condition.

Reply:
Its still correct later.
Even if we don't use it for multiply.

Reply:
Differentials actually should be required to define `zero` anyway.
Because of the vector space-ishness of them.

I just haven't got round to fixing a lot of that math and docs still.

Reply:
I will make a follow up PR if you don't want to.